src/Controller/CRUDController.php

```diff
@@ -368,15 +368,16 @@ public function batchAction(Request $request)
         if ($data = json_decode($request->get('data', ''), true)) {
             $action = $data['action'];
             $idx = $data['idx'];
-            $allElements = $data['all_elements'];
+            $allElements = (bool) $data['all_elements'];
             $request->request->replace(array_merge($request->request->all(), $data));
         } else {
             $request->request->set('idx', $request->get('idx', []));
-            $request->request->set('all_elements', $request->get('all_elements', false));
+
+            $allElements = $request->request->getBoolean('all_elements');
+            $request->request->set('all_elements', $allElements);
```

Comment:
`$request->request->set('all_elements', $allElements);` Is this needed?

Reply:
Maybe is in another parameter bag?

Reply:
Taking a look at https://github.com/sonata-project/SonataAdminBundle/commit/9949121fc0a475aa1547ed856593d00e171dabd8#diff-b1f1981b1ed9490b32c9bc83360655c3R396 where all this started, looks like it started using `Request::get()` method and it has been like that since then, but taking into account that some lines above there is this check:

`if (Request::METHOD_POST !== $restMethod)`

I would say we can use in this block `$request->request` safely (batch actions are sent [through `POST`](https://github.com/sonata-project/SonataAdminBundle/blob/3.x/src/Resources/views/CRUD/base_list.html.twig#L37)).

Reply:
Oké is this meaning that the line `$request->request->set('all_elements', $allElements);` (line 377) can be removed?

Reply:
Maybe this is used to allow a combination of POST and GET parameters?

Reply:
> Maybe this is used to allow a combination of POST and GET parameters?

https://github.com/sonata-project/SonataAdminBundle/blob/3.x/src/Controller/CRUDController.php#L410, apparently only `POST`.

Reply:
But you can POST to a url with for example the url `/delete?all_elements=1`

Reply:
Yes, but in this case:
- `all_elements`: https://github.com/sonata-project/SonataAdminBundle/blob/3.x/src/Resources/views/CRUD/base_list.html.twig#L167
- `action`: https://github.com/sonata-project/SonataAdminBundle/blob/3.x/src/Resources/views/CRUD/base_list.html.twig#L172
- `idx`: https://github.com/sonata-project/SonataAdminBundle/blob/82a4d6ea1a6eab39b48c5b1fcf54eecf2e88dbfd/src/Resources/views/CRUD/list__batch.html.twig#L15

I don't think they are later modified by JavaScript and seeing the `<form>` tag: https://github.com/sonata-project/SonataAdminBundle/blob/3.x/src/Resources/views/CRUD/base_list.html.twig#L37 I would say that the URL only contains filter parameters. 

And looking at how the request is built in [CRUDControllerTest](https://github.com/sonata-project/SonataAdminBundle/blob/3.x/tests/Controller/CRUDControllerTest.php) when using `idx` or `all_elements`.

So I think we can change this to just use `$request->request`.